sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -126,8 +126,8 @@ object RaiseError {
   """,
   since = "2.0.0",
   group = "misc_funcs")
-case class AssertTrue(left: Expression, right: Expression, child: Expression)
-  extends RuntimeReplaceable {
+case class AssertTrue(left: Expression, right: Expression, replacement: Expression)
+  extends RuntimeReplaceableInheritingTypeCoercion {
```

Comment:
Do we need to remain such constructor for such cases?

Why not just like
```
case class AssertTrue(left: Expression, right: Expression)
  extends RuntimeReplaceableInheritingTypeCoercion {
    lazy val replacement: Expression = If(left, Literal(null), RaiseError(right))
```

Reply:
See the classdoc of `RuntimeReplaceableInheritingTypeCoercion`. `replacement` must be in the constructor, so that catalyst rules can transform it.

Reply:
> See the classdoc of `RuntimeReplaceableInheritingTypeCoercion`. `replacement` must be in the constructor, so that catalyst rules can transform it.

Got it, thanks